### PR TITLE
chore(l1): reduce unnecessary docker builds and remove duplicate workflow executions in main

### LINKED ISSUE #1008: Reuse docker build in Hive jobs.
Currently, we're building our docker image for each Hive job. Instead, we should reuse the image from `docker-build`. 

See: https://docs.docker.com/build/ci/github-actions/share-image-jobs/

Instead of using `make run-hive`, that depends on `build-image`, we should run hive directly.

### LINKED ISSUE #1041: Remove workflow execution duplication when merging on main
This is due to having both `merge_group` and `push` rules, we need to get rid of the push ones:

```yml
name: CI
on:
  merge_group:
  push:
    branches: ["main"]
  pull_request:
    branches: ["**"]
    paths-ignore:
      - "README.md"
      - "LICENSE"
      - "**/README.md"
      - "**/docs/**"
```

image

### PR DESCRIPTION
**Pull request by rodrigo-o**
**Motivation**

We are re-building our docker image across the jobs/workflows multiple times, the idea is to remove as much of the duplicate executions as possible. At the same time, while merging to main we run duplicated jobs, the idea is to remove that duplication.

**Description**

This PR accomplish a couple of things:
- It creates a new make tasks for running hive without depending on the image build
- It generates a new job in the hive workflow to build the image once and then share it with every simulation
- Merges the assertoor stability check with hive to leverage the docker image already built
- Fix an issue were workflows were triggered two times in a row while merging to main (both `merge_group` and `push` rules were set)

Resolves #1008
Resolves #1041

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Docker Image
```
Comment (mpaulucci) [inline .github/workflows/build.yaml:1]: should you name this workflow docker?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -73,20 +73,6 @@ jobs:
         run: |
```
Comment (mpaulucci) [inline .github/workflows/ci.yaml:73]: maybe rename workflow to build?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (rodrigo-o) [reply .github/workflows/build.yaml:1]: For sure! still fully WIP and with lots of bad naming just trying to make it work ^^

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (rodrigo-o) [reply .github/workflows/ci.yaml:73]: Yeah, naming is bad still

**Comment by rodrigo-o** [comment]
Unfortunately, I wasn't able to share the Docker image across workflows, the only way to depend on other workflows for the triggering condition [is pushing them first to the default branch](https://github.com/orgs/community/discussions/25288#discussioncomment-3247293), which I wasn't really convinced and preferred to just make it work. The caveat? both hive and assertoor now share the same workflow, this makes the naming a bit longer; apart from that the Docker build is also run in the CI (to let the L2/EVM teams still leverage the build as part of their checks\merge queue) timing shouldn't be an issue due to both run at the same time.

That said, we now removed 6 unnecessary builds from the run.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (rodrigo-o) [reply .github/workflows/build.yaml]: I removed the new files and commented about it here: https://github.com/lambdaclass/lambda_ethereum_rust/pull/1032#issuecomment-2450670401

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (rodrigo-o) [reply .github/workflows/ci.yaml]: As with the previous comment, I removed the new file, commented here: https://github.com/lambdaclass/lambda_ethereum_rust/pull/1032#issuecomment-2450670401

File: Makefile [hunk 3]
```diff
@@ -97,6 +97,9 @@ TEST_PATTERN ?= /
 run-hive: build-image setup-hive ## 🧪 Run Hive testing suite
 	cd hive && ./hive --sim $(SIMULATION) --client ethereumrust --sim.limit "$(TEST_PATTERN)"
 
+run-hive-on-latest: setup-hive ## 🧪 Run Hive testing suite with the latest docker image
```
Comment (fkrause98) [inline Makefile:100]: Why do we need this new make target? Looks identical to run-hive

Comment (rodrigo-o) [reply Makefile:100]: It is, the only difference is it doesn't build the image as a prereq for the task, which allows us to resuse the image we already built as part of the workflow.

**Review by fkrause98 — APPROVED** [review]

**Pull request merged**